kt/godot-runtime/src/main/kotlin/godot/runtime/Registration.kt

```diff
@@ -89,18 +89,19 @@ class ClassBuilderDsl<T : KtObject>(
         }
 
         properties[propertyName] = KtProperty(
-            KtPropertyInfo(
-                KtVariant.Type.LONG,
-                propertyName,
-                "Int",
-                PropertyHint.ENUM,
-                enumValues<P>().joinToString { it.name }
-            ),
-            kProperty,
-            { enum -> KtVariant(enum.ordinal) },
-            { ktVariant -> enumValues<P>()[ktVariant.asInt()] },
-            KtVariant(Unit),
-            false
+                KtPropertyInfo(
+                        VariantType.LONG,
+                        propertyName,
+                        "Int",
+                        PropertyHint.ENUM,
+                        enumValues<P>().joinToString { it.name }
+                ),
+                kProperty,
+                { enum -> getVariantType(enum.ordinal) },
+                { any -> enumValues<P>()[(any as Long).toInt()] },
+                //TODO : Not sure here
+                enumValues<P>()[0],
```

Comment:
Still relevant?
I would use `null` as no value was set. Using the first enum value could lead to undesired behaviour no?
Also it would match with what we had with `KtVariant(Unit)`

Reply:
`null` default value is not supported for now IIRC.
I don't think we can even register a nullable type for now.

Reply:
Hmm we definitely have to support that i think. But let's do it in a separate PR then. Your Todo is fine for me :-)